File: frontend/demo/component/checkbox/checkbox-trigger.ts
import 'Frontend/demo/init'; // hidden-source-line

import { html, LitElement } from 'lit';
import { customElement } from 'lit/decorators.js';
import '@vaadin/vaadin-checkbox/vaadin-checkbox';
import { applyTheme } from 'Frontend/generated/theme';

// tag::snippet[]
@customElement('checkbox-trigger')
export class Example extends LitElement {
  protected createRenderRoot() {
    const root = super.createRenderRoot();
    // Apply custom theme (only supported if your app uses one)
    applyTheme(root);
    return root;
  }

  render() {
    return html`<vaadin-checkbox>Enabled</vaadin-checkbox>`;
  }
}
// end::snippet[]

File: src/main/java/com/vaadin/demo/component/checkbox/CheckboxTrigger.java
package com.vaadin.demo.component.checkbox;

import com.vaadin.demo.DemoExporter; // hidden-source-line
import com.vaadin.flow.component.checkbox.Checkbox;
import com.vaadin.flow.component.html.Div;
import com.vaadin.flow.router.Route;

@Route("checkbox-trigger")
public class CheckboxTrigger extends Div {

    public CheckboxTrigger() {
        // tag::snippet[]
        Checkbox checkbox = new Checkbox();
        checkbox.setLabel("Enabled");

        add(checkbox);
        // end::snippet[]
    }

    public static class Exporter extends DemoExporter<CheckboxTrigger> { // hidden-source-line
    } // hidden-source-line
}
